app/controllers/boxes_controller.rb

```diff
@@ -96,12 +104,46 @@ def load_box
     @box = Box.where(institution: @navigation_context.institution).find(params.fetch(:id))
   end
 
+  def load_box_print
+    begin
+      @box = Box.where(institution: @navigation_context.institution).find(params.fetch(:id))
+    rescue ActiveRecord::RecordNotFound
+      @transfer_package = TransferPackage.where("id = ? and ( sender_institution_id = ? or receiver_institution_id = ? )", params.fetch(:transfer_package), @navigation_context.institution['id'], @navigation_context.institution['id']).first
+      @box = Box
+        .joins(box_transfers: :transfer_package)
+        .where( transfer_packages: { id: @transfer_package.id })
+        .find(params.fetch(:id))
+    end
+  end
+
   def load_box_samples
     samples = @box.samples.preload(:batch, :sample_identifiers)
-    samples = samples.scrambled if @box.blinded?
     SamplePresenter.map(samples, request.format)
   end
 
+  def load_box_samples_print
+    samples = @box.samples.preload(:batch, :sample_identifiers)
+    if is_sender or !@box.blinded
```

Comment:
Checking a boolean is cheaper than running a SQL query, so I would test the other way around (it will only check the sender if the box is blinded):

```suggestion
    if !@box.blinded || is_sender
```